PFOffer.sol

```diff
@@ -98,6 +118,7 @@ contract PFOffer {
         oneTimeCosts = _oneTimeCosts;
         minDailyWithdrawLimit = _minDailyWithdrawLimit;
         dailyWithdrawLimit = _minDailyWithdrawLimit;
+        proposalID = _proposalID;
```

Comment:
Yes I have considered that. It's easy to get the proposal ID. It's just always `proposalsNumber()+1`.

The problem with this approach is that you gotta make the proposal right after you deploy the contract. And even when you do you may collide with someone else who is making a proposal at the same time.


I had not thought of a separate `watchContract` function. That's a nice idea. But we need to have a way to prevent the contractor from setting it to a proposalID that's not theirs in order to fool the `checkVoteStatus()` function.

Probably the easiest way to do so is by means of community awareness. If the proposalID does not match the one that actually tries to sign() the contract we can put out a community alert that the contractor is a scammer.